JS/display_article_js.js

```diff
@@ -0,0 +1,134 @@
+     */
+    link_to_show_article.onclick = function (){
+        //Need to load the html page and to give to the function that displays the article the informations required
+        localStorage.setItem('name_article', article.name);
+        localStorage.setItem('description_article', article.description);
+        localStorage.setItem('price_article', article.price);
+        localStorage.setItem('brand_article', article.brand);
+        localStorage.setItem("image_article", article.preview);
+        localStorage.setItem("id_article", article.id);                                      
+    }
+    html_component_article_info_div.appendChild(html_component_article_name);
+    html_component_article_info_div.appendChild(html_component_article_brand);
+    html_component_article_info_div.appendChild(html_component_article_price);
+
+    /*Add the values of the article in the component made for it:
+    * - article.price → html_component_article_price
+    * - article.name → html_component_article_name
+    * - article.brand → html_component_article_brand
+    */
+    let article_price = document.createTextNode((parseFloat(article.price)).toFixed(2) + "€");
+    html_component_article_price.appendChild(article_price);
+
+    let article_name = document.createTextNode(article.name);
+    html_component_article_name.appendChild(article_name);
+
+    let article_brand = document.createTextNode(article.brand);
+    html_component_article_brand.appendChild(article_brand);
+    html_component_article_image.src = article.preview;
+    html_component_article_image.id = "image_of_article_in_display";
+
+//pour ne pas avoir à passer par le csv
+//    return html_component_div_container;
+    div_container_all_article.appendChild(html_component_div_container);
+}
+
+//We create a XMLHttpRequest to read the articles.
+let httpRequest = new XMLHttpRequest();
+httpRequest.onreadystatechange = function () {
+    //readyState = 4 means that the operation with the XMLHttpRequest is done.
+    if (this.readyState === 4) {
+        // status = 200 mean that the statut of the XMLHttpRequest is OK (everything is going well)
+        if (this.status == 200) {
+            result_http_request = JSON.parse(this.responseText);
+            //For each element, the code add dynamically with the function create_list_article the article to the component "div_container_all_article"
+            for (let i = 0; i < result_http_request.length; i++) {
+                div_container_all_article.appendChild(
+                    create_list_article(result_http_request[i])
+            );
+        }
+        } else {
+            console.log("fail");
+        }
+    }
+};
+// The code take the data from a json file saved on github.
+httpRequest.open(
+    "GET",
+    "https://raw.githubusercontent.com/Yasmarila/Site_web_test/main/test?token=GHSAT0AAAAAABUG3RZ5TJOUIGOYXJ6V4ILYYT36LXA",
```

Comment:
I hope for you this token is not a github acces token, if so hurry up and remove it from your account ....